### ISSUE TITLE
train_model.py

### ISSUE THREAD

**Issue by Dh110b** [Dh110b/Dharshan#3]
import os
import cv2
import numpy as np
from sklearn.model_selection import train_test_split
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.callbacks import EarlyStopping  

data = []
labels = []
signs = ["0","1","2","3","4","5","6","7","8","9"] 

for sign in signs:
    for filename in os.listdir(f"preprocessed_{sign}"):
        if filename.endswith((".jpg", ".png")): 
            image_path = os.path.join(f"preprocessed_{sign}", filename)
            image = cv2.imread(image_path, cv2.IMREAD_GRAYSCALE)
            if image is not None:
                data.append(image)
                labels.append(sign)

label_mapping = {sign: i for i, sign in enumerate(signs)}
numerical_labels = [label_mapping[label] for label in labels]

data = np.array(data)
numerical_labels = np.array(numerical_labels)

data = data.reshape(-1, 64, 64, 1)

X_train, X_test, y_train, y_test = train_test_split(data, numerical_labels, test_size=0.2, random_state=42)

X_train = X_train / 255.0
X_test = X_test / 255.0

model = keras.Sequential(
    [
        keras.Input(shape=(64, 64, 1)),
        layers.Conv2D(32, kernel_size=(3, 3), activation="relu", padding="same"), 
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Conv2D(64, kernel_size=(3, 3), activation="relu", padding="same"), 
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Conv2D(128, kernel_size=(3, 3), activation="relu", padding="same"), 
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Flatten(),
        layers.Dropout(0.5),
        layers.Dense(len(signs), activation="softmax"),
    ]
)

model.compile(loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"])
model.summary()

early_stopping = EarlyStopping(monitor='val_loss', patience=5, restore_best_weights=True)  

model.fit(
    X_train,
    y_train,
    batch_size=32,
    epochs=20,  
    validation_split=0.1,
    callbacks=[early_stopping], 
)

loss, accuracy = model.evaluate(X_test, y_test, verbose=0)
print(f"Test Loss: {loss:.4f}")
print(f"Test Accuracy: {accuracy:.4f}")

model.save("sign_language_model.h5")
print("Model saved successfully!")
